test/e2e/positive.go

```diff
@@ -50,4 +71,98 @@ var _ = ginkgo.Describe("[PositiveFlow] Deployer validation", func() {
 			gomega.Expect(vo.Errors).To(gomega.BeEmpty())
 		})
 	})
+
+	ginkgo.Context("with cluster with all the deployed components", func() {
+
+		ginkgo.It("should have API components", func() {
+			mf, err := api.GetManifests()
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+			key := getObjKey(mf.Crd.Name, mf.Crd.Namespace)
+			err = hp.WaitForObjectToBeCreated(key, &v1.CustomResourceDefinition{})
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+		})
+
+		ginkgo.It("should have all RTE components", func() {
+			mf, err := rte.GetManifests()
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+			mf = mf.UpdateNamespace()
+
+			for _, obj := range mf.ToObjects() {
+				key := getObjKey(obj.GetName(), obj.GetNamespace())
+				err = hp.WaitForObjectToBeCreated(key, obj)
+				gomega.Expect(err).ToNot(gomega.HaveOccurred(), fmt.Sprintf("object of kind %v was not created", obj.GetObjectKind()))
+			}
+		})
+
+		ginkgo.It("should have SCHED components", func() {
+			mf, err := sched.GetManifests()
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+			mf = mf.UpdateNamespace()
+
+			for _, obj := range mf.ToObjects() {
+				key := getObjKey(obj.GetName(), obj.GetNamespace())
+				err = hp.WaitForObjectToBeCreated(key, obj)
+				gomega.Expect(err).ToNot(gomega.HaveOccurred(), fmt.Sprintf("object of kind %v was not created", obj.GetObjectKind()))
+			}
+
+		})
+
+		ginkgo.It("should have all pods running in all namespaces", func() {
+			ginkgo.By("checking that resource-topology-exporter pod is running")
+			mf, err := rte.GetManifests()
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+			mf = mf.UpdateNamespace()
+
+			found := getPodByRegex(fmt.Sprintf("%s-*", mf.DaemonSet.Name))
```

Comment:
another option is to take a list of pods in the cluster before `deploy()`, take another after `deploy()` completes succesfully and check the status of the pods found in the latter but not in the former.